src/scaraplate/cookiecutter.py

```diff
@@ -38,7 +40,7 @@ def __init__(self, target_path: Path) -> None:
         self.target_path = target_path
 
     @abc.abstractmethod
-    def read(self) -> Dict[str, str]:
+    def read(self) -> Dict[str, Any]:
```

Comment:
As cookiecutter supports [dict values](https://cookiecutter.readthedocs.io/en/latest/advanced/dict_variables.html) one can expect that Scaraplate does too.

I updated the YAMLConf docstring to : 
```python
{%- for key, value in cookiecutter.items()|sort %}
{%- if value is mapping %}
{{ key }}: {{ value }}
{%- else %}
{{ key }}: "{{ value }}"
{%- endif %}
{%- endfor %}
```
and set the `read` method return type to `Dict[str, Union[Dict, str]]` 
to support dict values while preserving.
Unfortunately I can't set this `Dict[str, Union[Dict, str]]`  return type on the `CookieCutterContext.read` method because of [this](https://github.com/python/mypy/issues/2300).

Reply:
> As cookiecutter supports [dict values](https://cookiecutter.readthedocs.io/en/latest/advanced/dict_variables.html) one can expect that Scaraplate does too.

Interesting, I didn’t know that. Do you have any idea how these dicts can be provided when entering values interactively via cli?

> and set the `read` method return type to `Dict[str, Union[Dict, str]]`
> to support dict values while preserving.

I guess this type should also allow lists (and, in fact, be recursive, which is not supported by mypy yet). If we choose to allow dicts, I think it would be okay to stick with `Dict[str, Any]`. But the bigger problem would be adding support for dicts to the already existing configparser-based implementations, which don’t provide support for arbitrary data types.